### ISSUE TITLE
City modal

### ISSUE THREAD

**Issue by yashha** [Rostlab/JS16_ProjectF#132]
Should the city modal in the map contain any information?
or an image?

**Comment by yashha** [comment]
maybe sth like this?
http://awoiaf.westeros.org/index.php?title=Raventree_Hall&printable=yes

**Comment by gyachdav** [comment]
yeah, although we havent seen this functionality in a while now it should show information pulled from the wiki. this was thoroughly discussed here https://github.com/Rostlab/JS16_ProjectC_Group10/issues/186 

@mbandle @AlexBeischl @tPiffrader please chime in

**Comment by yashha** [comment]
Inclusion of wiki works for me

**Comment by gyachdav** [comment]
as in you fixed it? or it just works for you as is?

**Comment by yashha** [comment]
http://tinypic.com/r/2lvxw28/9

**Comment by gyachdav** [comment]
looks awful :smile: I remember @mbandle showing a much nicer integration with the wiki.

**Comment by gyachdav** [comment]
I really think there was just a callback that you needed to use to get the correct content pulled into this div and laid out nicely. @mbandle said the callbacks are described here https://github.com/Rostlab/JS16_ProjectC_Group10/wiki/Init-New-GoT-Map-and-Options

**Comment by yashha** [comment]
I just wanted to show the functionality

Shell we make a completly new ui with the information given in the in the callback?

**Comment by gyachdav** [comment]
no no, I remember they actually pulled in content from the wiki and showed it in the modal. Let's wait for them to respond before you start making new UIs. The solution to this should be very very simple.

**Comment by jorjo1** [comment]
Hey!
Project C had a very nice representation of the wiki for each character when opened on the map. As far as I can tell they used a simple iframe, am I right?

**Comment by sacdallago** [comment]
Yes. That was the case. @AlexBeischl @mbandle some yelp here?

**Comment by bandle** [comment]
We have used an Ajax request to the wiki page and parsed the outcome via jQuery to get only the relevant parts and include them into the modal. This doesn't work online because the wiki guys prohibited that their site is used via Ajax Requests by setting the CORS Headers: https://en.wikipedia.org/wiki/Cross-origin_resource_sharing

So iframe would possibly work in the modal. If you want to have a look at the code generating the modal content: Its in gotmap.js https://github.com/Rostlab/JS16_ProjectC_Group10/blob/develop/mockup/js/gotmap.js#L382

**Comment by gyachdav** [comment]
@yashha given the info we now have from @mbandle when do you expect this issue to be resolved? we also have #124 to finish here before we can call the map component fully integrated. Please assign someone from your team to take care of this ASAP. This is now high priority.

**Comment by gyachdav** [comment]
@AlexBeischl @mbandle @tPiffrader please see this to completion (i.e. make sure that you get all the functionality actually working on the web app as expected).

**Comment by mammuth** [comment]
So beside the iFrame solution there would be a hacky one aswell I guess. The iframe solution should be straight forward, but well... then we have iframes.

The hacky solution:
CORS is just a security method in the browsers. So what should work is to query some script that is executed by our webserver on the modals-callback, like: 

```
jQuery.ajax({
  url: '/wikirequest/'+city
}) 
```

Where the script at /wikirequest/ might be eg. a php script that is executed by nginx or a js script that runs inside node and returns the wiki-page as response to the ajax call.

This way we could make use of the nice wiki-parsing that the map-guys already did (and doesn't work anymore due to the CORS headers).

**Comment by sacdallago** [comment]
I like no iframes :D :D :D let's see how much this can be done.

Last resort is to redirect on a new tab to the corresponding wiki page. Ugly AF. Again: **LAST RESORT**

**Comment by mammuth** [comment]
I got it running by using my suggestion above. :+1: 
Will create a PR later when I was able to ask some questions about our server-component and the webpack-setup that are related to my implementation.

<img width="1680" alt="screen shot 2016-03-26 at 15 11 15" src="https://cloud.githubusercontent.com/assets/3121306/14060483/ad7f0ad8-f365-11e5-919e-94875b3b4b00.png">

**Comment by sacdallago** [comment]
Might not look as fancy, but for me it works. @gyachdav ?

**Comment by gyachdav** [comment]
which branch should i switch to, to see this working?

**Comment by mammuth** [comment]
city-modal. 
Since the dev-server seems to be broken, you need to build and serve it: npm run build, npm run serve.

**Comment by gyachdav** [comment]
btw, @mbandle in the version we saw a week+ ago the modal was showing the details from the wiki plus an image of a map from the wiki. did you guys pulled any specific divs, images from the awoiaf wiki when showing this modal?

**Comment by mammuth** [comment]
In the modal that is working now, I removed all images from the pulled wiki-page, because the browser won't be able to show them due to the wiki's CORS-Headers.

**Comment by gyachdav** [comment]
Doesn't work on my local instance :worried: 
![screen shot 2016-03-26 at 5 15 25 pm](https://cloud.githubusercontent.com/assets/4074881/14062468/860705e4-f376-11e5-8db3-8d459e3138b8.png)

also clicking on the Show in Wiki button generates this url http://awoiaf.westeros.org/index.php/[object%20Object] which is clearly malformed

**Comment by gyachdav** [comment]
oops clicked wrong button- reopenning

**Comment by gyachdav** [comment]
btw i did:
1. npm install
2. npm start 

as @yashha  suggested.

**Comment by mammuth** [comment]
@gyachdav I guess you overlooked my note above, that the dev-server seems not to work right now.

You can test the modal by building the app:
- npm run build
- npm run serve 

The link to the wiki was a bug, thanks :+1:

**Comment by gyachdav** [comment]
yup I did try the build procedure you suggested but got this:

http://pastebin.com/4vHiBuMH

seems that the build process fails on: 

 Twit config must include `consumer_key` when using user auth.

**Comment by yashha** [comment]
You still have to 
1. duplicate the config.default.json to config.json
2. add twitter access keys to gotsent in the config
3. add mongodb access url to gotsent.mongo.uri  in the config (I will send it to you by email)

Or wait untill we stage.
It is currently stageable.

**Comment by gyachdav** [comment]
k ill wait then. thanks!

**Comment by yashha** [comment]
No problem :)
Ill probably have to explain the steps in more detail in the readme

**Comment by mammuth** [comment]
Merged into dev

**Comment by gyachdav** [comment]
Thanks. please stage.
